src/v/archival/ntp_archiver_service.cc

```diff
@@ -2951,46 +2951,139 @@ size_t ntp_archiver::get_local_segment_size() const {
     return log_segment_size;
 }
 
-ss::future<bool>
-ntp_archiver::prepare_transfer_leadership(ss::lowres_clock::duration timeout) {
-    _paused = true;
-
+ss::future<bool> ntp_archiver::pause(ss::lowres_clock::duration timeout) {
```

Comment:
Why not just stop the archiver the same way we do during topic-configuration update?

Reply:
This multiplies number of state transitions by two and it will be used very rarely so it's a susceptible to cod rot.

Reply:
Yeah, I tried what you suggested and it turned out a lot simpler. Will update